archunit/src/main/java/com/tngtech/archunit/core/domain/TypeCast.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2014-2022 TNG Technology Consulting GmbH
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.tngtech.archunit.core.domain;
+
+import com.tngtech.archunit.PublicAPI;
+import com.tngtech.archunit.core.domain.properties.HasOwner;
+import com.tngtech.archunit.core.domain.properties.HasSourceCodeLocation;
+import com.tngtech.archunit.core.domain.properties.HasType;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkNotNull;
+import static com.tngtech.archunit.PublicAPI.Usage.ACCESS;
+
+@PublicAPI(usage = ACCESS)
+public final class TypeCast implements HasType, HasOwner<JavaCodeUnit>, HasSourceCodeLocation {
+
+    private final JavaCodeUnit owner;
+    private final JavaClass value;
+    private final int lineNumber;
+    private final SourceCodeLocation sourceCodeLocation;
+
+    private TypeCast(JavaCodeUnit owner, JavaClass value, int lineNumber) {
+        this.owner = checkNotNull(owner);
+        this.value = checkNotNull(value);
+        this.lineNumber = lineNumber;
```

Comment:
We don't have to store the `lineNumber` redundantly, but can always use `sourceCodeLocation.getLineNumber()`.

Reply:
ok i will not store lineNumber